### ISSUE TITLE
zero length arrays don't work

### ISSUE THREAD

**Issue by sapir** [sapir/llvm-cbe#1]

**Comment by vtjnash** [comment]
I think the compiler warnings around them may have changed recently? (they've always been disallowed by the standard, so can be annoying to handle)

**Comment by sapir** [comment]
Well, I'm used to using them anyway :). I see now that they're not standard (https://stackoverflow.com/questions/11733981/what-is-the-purpose-of-a-zero-length-array-in-a-struct) so maybe my tests should be testing a "flexible array member" as described in the link, instead.

But really I'm trying to make the CBE work well enough to compile Rust to LLVM IR to C, and I think that I added this after running into a "zero-length array" in some of the LLVM IR output from rustc.

**Comment by vtjnash** [comment]
Agreed. Not quite the same as a flexible array member though. Rather annoying to emit correctly though (since the edge case works fine in Julia, LLVM, and Rust, but is undefined in C and wrong in C++) and hard to test. Generally it would be ok just to drop and ignore this element, but that means the struct gep indexes need to be recalculated.
